flask_admin/contrib/fileadmin/azure.py

```diff
@@ -1,13 +1,19 @@
 import os.path as op
 from datetime import datetime
 from datetime import timedelta
-from time import sleep
 
 try:
-    from azure.storage.blob import BlobPermissions
-    from azure.storage.blob import BlockBlobService
-except ImportError:
-    BlobPermissions = BlockBlobService = None
+    from azure.core.exceptions import ResourceExistsError
+    from azure.storage.blob import BlobProperties
+    from azure.storage.blob import BlobSasPermissions
+    from azure.storage.blob import BlobServiceClient
+    from azure.storage.blob import generate_blob_sas
+except ImportError as e:
+    raise Exception(
```

Comment:
I changed the handling of the importerror to make mypy happier. I looked at other optional modules and they all seemed to handle importerrors a bit differently. I found one that did it this way. This seems fine since you should import the module unless you're using it.